gobblin-data-management/src/main/java/gobblin/data/management/copy/entities/CommitStepDB.java

```diff
@@ -0,0 +1,198 @@
+/*
+ * Copyright (C) 2014-2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.data.management.copy.entities;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.SQLIntegrityConstraintViolationException;
+import java.sql.Statement;
+import java.util.NoSuchElementException;
+import java.util.Properties;
+
+import org.apache.commons.dbcp2.BasicDataSource;
+import org.apache.commons.io.IOUtils;
+
+import com.google.common.base.Charsets;
+
+import gobblin.commit.CommitStep;
+import gobblin.data.management.copy.CopyEntity;
+
+import lombok.AccessLevel;
+import lombok.NoArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+
+/**
+ * A DB that stores {@link CommitStep}s. Implementation is transparent to the user, but currently uses Derby.
+ */
+@Slf4j
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+public class CommitStepDB {
+
+  public static final String COMMIT_STEP_DB_NAME = "gobblinCommitStepDB";
+  private static final String PROTOCOL = "jdbc:derby:";
+  private static final String ALREADY_EXISTS_STATE = "X0Y32";
+
+  private static final String INSERT_STEP_STATEMENT = "INSERT INTO steps VALUES (?, ?)";
+  private static final String GET_STEP_STATEMENT = "SELECT step FROM steps WHERE step_key = ?";
+
+  private static final BasicDataSource BDS = new BasicDataSource();
+  static {
+    initialize();
+  }
+
+  private static void initialize() {
```

Comment:
As discussed, shared static state is bad as it may lead to unpredictable results if I decide to launch two gobblin instances.